core-primitives/sgx/crypto/src/key_repository.rs

```diff
@@ -62,6 +72,19 @@ where
 	}
 }
 
+impl<Pair, SealedIo> AccessPubkey for KeyRepository<Pair, SealedIo>
+where
+	Pair: ToPubkey<Error = crate::error::Error> + Clone,
+	SealedIo: SealedIO<Unsealed = Pair, Error = crate::error::Error>,
+{
+	type KeyType = <Pair as ToPubkey>::Pubkey;
+
+	fn retrieve_pubkey(&self) -> Result<Self::KeyType> {
+		let pair = self.key_lock.read().map_err(|_| Error::LockPoisoning).map(|l| l.clone())?;
```

Comment:
cant we get same desire behavior with:
```self.key_lock.read().map_err(|_| Error::LockPoisoning)?.clone();```

Reply:
Yes, you are right. I blindly copied and pasted that code XD.

Reply:
Thanks to you, I noticed that the clone was not necessary whatsoever.